internal/extproc/server.go

```diff
@@ -125,6 +160,17 @@ func (s *Server[P]) process(p P, stream extprocv3.ExternalProcessor_ProcessServe
 			return status.Errorf(codes.Unknown, "cannot receive stream request: %v", err)
 		}
 
+		// If we're processing the request headers, read the :path header to instantiate the
+		// right processor.
+		if headers := req.GetRequestHeaders().GetHeaders(); headers != nil {
```

Comment:
This is the main change in this PR.

In order to allow having custom processors per-request path we need to access the `:path` header. We can't instantiate the `ProcessorIface` object until we've read the first request headers message. This is why I instantiate the processor here, and reuse it for the following messages.